rust/osm2lanes/src/transform/lanes_to_tags.rs

```diff
@@ -219,23 +221,33 @@ pub fn lanes_to_tags(lanes: &[Lane], locale: &Locale, config: &LanesToTagsConfig
         tags.checked_insert("turn:lanes:both_ways", "left")?;
     }
 
-    let max_speeds: Vec<Speed> = lanes
-        .iter()
-        .filter_map(|lane| match lane {
-            Lane::Travel { max_speed, .. } => *max_speed,
-            _ => None,
-        })
-        .collect();
-    if let Some(max_speed) = max_speeds.first() {
-        if max_speeds.windows(2).all(|w| w[0] == w[1]) {
-            tags.checked_insert("maxspeed", max_speed.to_string())?;
-        } else {
-            return Err(RoadMsg::Unimplemented {
-                description: Some("different max speeds per lane".to_owned()),
-                tags: None,
+    let max_speed = {
+        let max_speeds: Vec<Speed> = lanes
```

Comment:
If you collect into `HashSet`, you can do `.len()` below instead of checking every pair -- seems simpler?

Reply:
this would be a fun thing to microbench. Will set it up